src/vs/workbench/contrib/terminal/electron-browser/terminalInstance.ts

```diff
@@ -212,8 +212,8 @@ export class TerminalInstance implements ITerminalInstance {
 	public get shellLaunchConfig(): IShellLaunchConfig { return this._shellLaunchConfig; }
 	public get commandTracker(): TerminalCommandTracker { return this._commandTracker; }
 
-	private readonly _onExit = new Emitter<number>();
-	public get onExit(): Event<number> { return this._onExit.event; }
+	private readonly _onExit = new Emitter<number | undefined>();
```

Comment:
I was initially puzzled by this but it turns out this is valid and out strict null checks aren't working correctly here. `TerminalProcess._exitCode` is actually `number | undefined` but somehow that's slipping through the type checker. Tasks relies on `TerminalProcess._exitCode` being `undefined` to tell whether the dispose was a user action.

Reply:
@Tyriar - Is there something you'd like me to address?

Reply:
This is more of a comment as it's not clear from the code.